BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -49,22 +49,70 @@ public actor DefaultTabDataStore: TabDataStore {
     // MARK: Fetching Window Data
 
     public func fetchWindowData(uuid: UUID) async -> WindowData? {
-        logger.log("Attempting to fetch window/tab data", level: .debug, category: .tabs)
+        logger.log("Attempting to fetch window data", level: .debug, category: .tabs)
+
+        // Adding more logging for FXIOS-9517
+        var shouldLogFileFailure = false // Whether pulling from the main file failed
+        var shouldLogBackupFailure = false // Whether pulling from the backup file failed
+        var fileInfoMessage = "" // Specifics of main file failure
+        var backupInfoMessage = "" // Specifics of backup file failure
+        defer {
+            if shouldLogFileFailure {
```

Comment:
Super minor nit: we could use a guard here to avoid an extra level of identation / `if`